src/clib/lib/enkf/enkf_config_node.cpp

```diff
@@ -691,29 +666,24 @@ enkf_config_node_type *enkf_config_node_alloc_GEN_DATA_from_config(
     return config_node;
 }
 
-enkf_config_node_type *enkf_config_node_alloc_GEN_DATA_full(
-    const char *node_key, const char *result_file,
-    gen_data_file_format_type input_format, const int_vector_type *report_steps,
-    const char *ecl_file, const char *init_file_fmt, const char *template_file,
-    const char *data_key) {
+enkf_config_node_type *
+enkf_config_node_alloc_GEN_DATA_full(const char *node_key,
+                                     const char *result_file,
+                                     gen_data_file_format_type input_format,
+                                     const int_vector_type *report_steps) {
     enkf_config_node_type *config_node = NULL;
 
-    if ((init_file_fmt == NULL) && (ecl_file == NULL) &&
-        (result_file != NULL)) {
+    if (result_file != NULL) {
         config_node = enkf_config_node_alloc_GEN_DATA_result(
             node_key, input_format, result_file);
-    } else if ((init_file_fmt != NULL) && (ecl_file != NULL) &&
-               (result_file != NULL)) {
+    } else if (result_file != NULL) {
```

Comment:
This does not make a lot of sense now that `init_file_fmt` and  `ecl_file` have been removed